src/ansiblelint/utils.py

```diff
@@ -901,3 +910,85 @@ def nested_items(
             yield "list-item", item, parent
             for k, v, p in nested_items(item):
                 yield k, v, p
+
+
+def nested_items_path(
+    data: Union[Dict[Any, Any], List[Any]],
+    parent_path: Optional[List[Union[str, int]]] = None,
+) -> Iterator[Tuple[Any, Any, List[Union[str, int]]]]:
+    """Iterate a nested data structure, yielding key/index, value, and parent_path.
+
+    This is a recursive function that calls itself for each nested layer of data.
+    Each iteration yields:
+
+    1. the current item's dictionary key or list index,
+    2. the current item's value, and
+    3. the path to the current item from the outermost data structure.
+
+    For dicts, the yielded (1) key and (2) value are what mydict.items() yields.
+    For lists, the yielded (1) index and (2) value are what enumerate(mylist) yields.
```

Comment:
I wouldn't use the invented my-prefixed identifiers as they add go the confusion, not being declared anywhere.
```suggestion
    For dicts, the yielded (1) key and (2) value are what ``dict.items()`` yields.
    For lists, the yielded (1) index and (2) value are what ``enumerate()`` yields.
```